apps/ledger-live-desktop/src/renderer/families/algorand/OptInFlowModal/steps/StepConfirmation.tsx

```diff
@@ -22,12 +24,12 @@ const Container = styled(Box).attrs(() => ({
 }>`
   justify-content: ${p => (p.shouldSpace ? "space-between" : "center")};
 `;
-function StepConfirmation({ account, optimisticOperation, error, signed, transaction }: StepProps) {
-  const options = account && listTokensForCryptoCurrency(account.currency);
-  const token = useMemo(
-    () => transaction && options && options.find(({ id }) => id === transaction.assetId),
-    [options, transaction],
-  );
+
+function StepConfirmation({ optimisticOperation, error, signed, transaction }: StepProps) {
+  invariant(transaction, "Transaction should be present");
```

Comment:
Small detail: While I like adding invariant calls in our code to make sure it's fine, we also need to add [error boundaries](http://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) where it make sense in the UI instead of having the fullscreen error we have at the moment IMO